In [123]:
import pandas as pd
import re

In [124]:
df_old = pd.read_csv('data_results/old_cross_val_results.csv')
df_new = pd.read_csv('data_results/post_tuning_cross_val_results.csv')

In [125]:
metrics = ['train_accuracy_score_mean', 'valid_accuracy_score_mean', 'train_accuracy_score_std', 'valid_accuracy_score_std' ]

In [126]:
df_old = df_old[['model'] + metrics]
df_old.model = df_old.model.apply(lambda text: text.lower().split('(')[0].replace('classifier',''))

df_new = df_new[['model'] + metrics]
df_new.model = df_new.model.apply(lambda text: text.lower().split('(')[0].replace('classifier',''))


In [128]:
df_new.columns = ['model', 'train accuracy post tuning', 'valid accuracy post tuning', 'train std post tuning', 'valid std post tuning']
df_new = df_new.sort_values('model')


In [129]:
df_old.columns = ['model', 'train accuracy before tuning', 'valid accuracy before tuning', 'train std before tuning', 'valid std before tuning']
df_old = df_old.sort_values('model')


In [130]:
df_acc = pd.DataFrame()
df_acc['model'] = df_new['model'].to_numpy()
df_acc['train accuracy before tuning'] = df_old['train accuracy before tuning'].apply(lambda acc: round(acc,4)).to_numpy()
df_acc['train accuracy post tuning'] = df_new['train accuracy post tuning'].apply(lambda acc: round(acc,4)).to_numpy()
df_acc['train trend'] = [u"\u2197", u"\u2198", u"\u2198", u"\u2198", u"\u2197",u"\u2197"]

df_acc['valid accuracy before tuning'] = df_old['valid accuracy before tuning'].apply(lambda acc: round(acc,4)).to_numpy()
df_acc['valid accuracy post tuning'] = df_new['valid accuracy post tuning'].apply(lambda acc: round(acc,4)).to_numpy()
df_acc['valid trend'] = [u"\u2197", u"\u2197", u"\u2197", u"\u2198", u"\u2197", u"\u2197"]


In [131]:
[u"\u2197",u"\u2198",u"\u2192"]

['↗', '↘', '→']

In [132]:
df_acc['valid accuracy post tuning'].to_numpy() - df_acc['valid accuracy before tuning'].to_numpy()

array([ 0.0018,  0.0033,  0.0006, -0.0025,  0.0005,  0.0136])

In [133]:
df_acc

model  train accuracy before tuning  \
0        decisiontree                        0.7864   
1           linearsvc                        0.9877   
2  logisticregression                        0.9301   
3                 mlp                        1.0000   
4       multinomialnb                        0.9100   
5        randomforest                        0.8789   

   train accuracy post tuning train trend  valid accuracy before tuning  \
0                      0.8087           ↗                        0.7340   
1                      0.9531           ↘                        0.8908   
2                      0.9275           ↘                        0.8898   
3                      0.8910           ↘                        0.8764   
4                      0.9157           ↗                        0.8599   
5                      0.9037           ↗                        0.8292   

   valid accuracy post tuning valid trend  
0                      0.7358           ↗  
1                      0.8941           ↗  
2                      0.8904           ↗  
3                      0.8739           ↘  
4                      0.8604           ↗  
5                      0.8428           ↗

- The first thing that can be noticed is that all the models have seen their validation accuracy increase except for the multilayer Perceptron. However, the increases remain relatively small, from 0.06% to 1.36%.
- The validation accuracy went from 87.64% to 87.39%, a loss of 0.25% of validation accuracy. However, the training accuracy went from 100% to 89%. We were thus able to keep a good average validation accuracy while suppressing the overfitting.
- The linear SVM and Logistic regression models, have slightly lost in average training accuracy but have still increased in average validation accuracy.
- The decision tree, multinomial Naive Bayes and random forest models have seen their average training and validation accuracy increase.


In [134]:
df_std = pd.DataFrame()
df_std['model'] = df_new['model'].to_numpy()
df_std['train std before tuning'] = df_old['train std before tuning'].apply(lambda acc: round(acc,5)).to_numpy()
df_std['train std post tuning'] = df_new['train std post tuning'].apply(lambda acc: round(acc,5)).to_numpy()
df_std['train trend'] = [u"\u2197", u"\u2197", u"\u2197", u"\u2197", u"\u2198",u"\u2198"]

df_std['valid std before tuning'] = df_old['valid std before tuning'].apply(lambda acc: round(acc,5)).to_numpy()
df_std['valid std post tuning'] = df_new['valid std post tuning'].apply(lambda acc: round(acc,5)).to_numpy()
df_std['valid trend'] = [u"\u2197", u"\u2198", u"\u2198", u"\u2198", u"\u2197", u"\u2197"]


In [135]:
df_std

model  train std before tuning  train std post tuning  \
0        decisiontree                  0.00193                0.00246   
1           linearsvc                  0.00037                0.00047   
2  logisticregression                  0.00060                0.00065   
3                 mlp                  0.00001                0.00479   
4       multinomialnb                  0.00101                0.00092   
5        randomforest                  0.00354                0.00147   

  train trend  valid std before tuning  valid std post tuning valid trend  
0           ↗                  0.00411                0.00486           ↗  
1           ↗                  0.00256                0.00195           ↘  
2           ↗                  0.00252                0.00234           ↘  
3           ↗                  0.00423                0.00352           ↘  
4           ↘                  0.00464                0.00526           ↗  
5           ↘                  0.00460                0.00718           ↗